report/4.Solution/SemanticsDescription.tex

```diff
@@ -0,0 +1,762 @@
+	\end{math}
+	
+	\texttt{ where } $env_V, sto \vdash e \rightarrow_e v$
+	\texttt{ and } $env_V\ x = l$
+\end{center}
+
+\texttt{[$Comp_{BSS}$]}\\
+\begin{center}
+	\begin{math}
+		\cfrac
+			{env_V, env_P \vdash \langle S_1, sto \rangle \rightarrow_{stm} sto'' \quad env_V, env_P \vdash \langle S_2, sto'' \rangle \rightarrow_{stm} sto'}
+			{env_V, env_P \vdash \langle S_1; S_2, sto \rangle \rightarrow_{sto'}}
+	\end{math}
+\end{center}
+
+\texttt{[$Then-True_{BSS}$]}\\
+\begin{center}
+	\begin{math}
+		\cfrac
+			{env_V, env_P \vdash \langle L,sto \rangle \rightarrow_{stm} sto'}
+			{env_V, env_P \vdash \langle b\ then\ L, sto \rangle \rightarrow_{stm} sto'}
+	\end{math}
+	
+	\texttt{ if } $env_V, sto \vdash b \rightarrow_b True$
+\end{center}
+
+\texttt{[$Then-False_{BSS}$]}\\
+\begin{center}
+	\begin{math}
+	\cfrac
+	{env_V, env_P \vdash \langle \epsilon,sto \rangle \rightarrow_{stm} sto}
+	{env_V, env_P \vdash \langle b\ then\ L, sto \rangle \rightarrow_{stm} sto}
+	\end{math}
+	
+	\texttt{ if } $env_V, sto \vdash b \rightarrow_b False$
+\end{center}
+
+\texttt{[$While-False_{BSS}$]}\\
+\begin{center}
+	\begin{math}
+		{env_V, env_P \vdash \langle L\ while\ b, sto \rangle \rightarrow_{stm} sto}
+	\end{math}
+	
+	\texttt{ if } $env_V, sto \vdash b \rightarrow_b False$
+\end{center}
+
+\texttt{[$While-True_{BSS}$]}\\
+\begin{center}
+	\begin{math}
+		\cfrac
+			{env_V, env_P \vdash \langle L, sto \rangle \rightarrow_{stm} sto'' \quad env_V, env_P \vdash \langle L\ while\ b, sto'' \rangle \rightarrow_{stm} sto'}
+			{env_V, env_P \vdash \langle L\ while\ b, sto \rangle \rightarrow_{stm} sto'}
+	\end{math}
+	
+	\texttt{ if } $env_V, sto \vdash b \rightarrow_b True$
+\end{center}
+
+\texttt{[$ProcCall-NoParams_{BSS}$]}\\
+\begin{center}
+	\begin{math}
+		\cfrac
+			{env_v'[x \vdash l][nest \vdash new\ l], env_P'[p \vdash (L, e, env_V', env_P')] \vdash \langle L, sto[l \vdash v] \rangle \rightarrow_{stm} sto'}
+			{env_v, env_P \vdash \langle p, sto \rangle \rightarrow_{stm} sto'}
+	\end{math}
+	
+	\texttt{ where } $env_P\ p = (L, e, env_V', env_P')$\\
+	\texttt{ and } $ env_V', sto \vdash e \rightarrow_{e} v$\\
+	\texttt{ and } $l = env_V\ next$
+\end{center}
+
+\texttt{[$ProcCall-Params_{BSS}$]}\\
+\begin{center}
+	\begin{math}
+	\cfrac
+	{env_v'[x \vdash l][nest \vdash new\ l], env_P'[p \vdash (L, x\sim, e, env_V', env_P')] \vdash \langle L, sto[l \vdash v] \rangle \rightarrow_{stm} sto'}
+	{env_v, env_P \vdash \langle x\ p\ x\sim, sto \rangle \rightarrow_{stm} sto'}
+	\end{math}
+	
+	\texttt{ where } $env_P\ p = (L, x\sim, e, env_V', env_P')$\\
+	\texttt{ and } $ env_V', sto \vdash e \rightarrow_{e} v$\\
+	\texttt{ and } $l = env_V\ next$
+\end{center}
+
+\texttt{[$VarDclStm_{BSS}$]}\\
+\begin{center}
+	\begin{math}
+		\cfrac
+			{\langle D_V, env_V, sto \rangle \rightarrow_{D_V} (env_V', sto')}
+			{env_V, env_P \vdash \langle D_V, sto \rangle \rightarrow_{stm} sto'}
```

Comment:
Fjern denne regel og den næste. For det første giver de ikke mening (den betyder lige nu: D_V ændrer på sto, hvis D_V ændrer på env_V og sto??? Det bliver endnu værre for D_P, som ikke ændrer på sto), men de er også unødvendige.
Jeg husker ikke, at Hans beskrev Bims/Bip/Bump med en Statement regel for D_V og D_P, så det synes jeg heller ikke, at vi skal gøre.

Reply:
Det er simpelthen fordi at en statement kan være en erklæring. Han gjorde det faktisk i SS det var blot igennem "begin end funktionen". Men nu da jeg kigger på den igen er de jo faktisk ret ligegyldige "tomme regler". Så jo vi fjerner dem bare ;)
Fixed